public/client-v2/src/components/FrontsEdit/Fronts.js

```diff
@@ -0,0 +1,84 @@
+// @flow
+
+import React from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import filterFrontsByPriority from '../../util/filterFrontsByPriority';
+import getFrontsConfig from '../../actions/FrontsConfig';
+import type { FrontDetail, FrontsConfig } from '../../types/Fronts';
+import type { State } from '../../types/State';
+
+type FrontsComponentProps = {
+  match: Object,
+  frontsActions: Object,
+  frontsConfig: FrontsConfig
+};
+
+type ComponentState = {
+  selectedFront: ?string
+};
+
+class Fronts extends React.Component<FrontsComponentProps, ComponentState> {
+  state = {
+    selectedFront: null
+  };
+
+  componentDidMount() {
+    this.props.frontsActions.getFrontsConfig();
+  }
+
+  renderSelectPrompt = () => {
+    if (!this.state.selectedFront) {
+      return <option value="">Choose a front...</option>;
+    }
+    return null;
+  };
+
+  renderFrontOption = (front: FrontDetail) => (
```

Comment:
Could this method be inline inside render?

Reply:
Maybe but I'm not too keen to shuffle this around now as this component should be getting more complicated quite soon.